Sources/Core/AST/Decl/AssociatedTypeDecl.swift

```diff
@@ -1,10 +1,10 @@
 /// An associated type declaration.
 public struct AssociatedTypeDecl: SingleEntityDecl {
 
-  public let origin: SourceRange?
+  public let origin: SourceRange
 
   /// The source range of the declaration's introducer, if any.
```

Comment:
```suggestion
  /// The source range of the declaration's introducer.
```

There's probably a lot of "...., if any" comments in the docs of all nodes.